scripts/igmp_query_injector.py

```diff
@@ -237,19 +100,19 @@ def __init__(self, max_resp_time):
 
     def inject_to_vif(self, vif, mac):
         log.info('Inject IGMP query to vif:%s, mac:%s' % (vif, mac))
-        packet = IGMPQueryGenerator(mac, -1, self.max_resp_time).generate()
-        self.inject_query_packet(vif, packet)
+        self.inject_packet(vif, mac)
 
     def inject_to_vifs(self, vifs):
         for vif in vifs:
             _vif = Vif(vif)
```

Comment:
Creating this object and just to read the vif_name and mac fields seems wasteful.

Reply:
replace with helper functions